src/ui/common/src/components/pages/artifact/id/index.tsx

```diff
@@ -0,0 +1,136 @@
+    //sourceLocation,
+}) => {
+    return (
+        <Box width="100%" display="flex" alignItems="center">
+            <FontAwesomeIcon height="24px" width="24px" style={{ marginRight: '8px' }} icon={faCircleCheck} color={'green'} />
+            <Typography variant="h4" component="div">
+                {artifactName}
+            </Typography>
+            {/* <Typography marginTop="4px" variant="caption" component="div">
+                Created: {createdAt}
+            </Typography>
+            <Typography variant="caption" component="div">
+                Source: <Link>{sourceLocation}</Link>
+            </Typography> */}
+        </Box>
+    );
+};
+
+type ArtifactDetailsPageProps = {
+    user: UserProfile;
+    Layout?: React.FC<LayoutProps>;
+};
+
+const ArtifactDetailsPage: React.FC<ArtifactDetailsPageProps> = ({
+    user,
+    Layout = DefaultLayout,
+}) => {
+    const dispatch: AppDispatch = useDispatch();
+    const { workflowDagResultId, artifactId } = useParams();
+    const artifactResult: ArtifactResult | null = useSelector(
+        (state: RootState) => {
+            // First, check if there are any keys in the artifactResults object.
+            const artifactResults = state.workflowReducer.artifactResults;
+            if (Object.keys(artifactResults).length < 1) {
+                return null;
+            }
+
+            return artifactResults[artifactId];
+        }
+    );
+
+    useEffect(() => {
+        document.title = 'Artifact | Aqueduct';
+
+        // Check and see if we are loading the artifact result
+        if (!artifactResult) {
+            dispatch(
+                handleGetArtifactResults({
+                    apiKey: user.apiKey,
+                    workflowDagResultId,
+                    artifactId,
+                })
+            );
+        }
+    }, []);
+
+    if (!artifactResult || !artifactResult.result) {
+        return (
+            <Layout user={user}>
+                <CircularProgress />
+            </Layout>
+        );
+    }
+
+    const parsedData = JSON.parse(artifactResult.result.data);
+
+    return (
+        <Layout user={user}>
+            <Box width={'800px'}>
+                <Box width="100%">
+                    <Box width="100%" display="flex">
+                        <ArtifactDetailsHeader
+                            artifactName={artifactResult.result.name}
+                        />
+                        <Button variant="contained" sx={{ maxHeight: '32px' }}>
+                            EXPORT
+                        </Button>
+                    </Box>
+                    <Box width="100%" marginTop="12px">
+                        <Typography variant="h5" component="div" marginBottom="8px">
+                            Preview
+                        </Typography>
+                        <StickyHeaderTable data={parsedData} />
+                    </Box>
+                    {/* <Box display="flex" width="100%" paddingTop="40px">
```

Comment:
The KeyValueTable component here is meant to be a component to easily display the results of metrics and checks. 

I'm not exactly sure how to go about getting the associated metrics and Checks for an ArtifactResult. Could use a little help here.